### PR TITLE
Deal with the sprig library using int for sub and div values

### LINKED ISSUE #71: PodDisruptionBudget calculation is not working?
I have a 3 replicas setup and for some reason the PDB was set to 0.

`ceil (sub (div (int .Values.server.replicas) 2) 1)` looks good to me but it's possible the calculation is off in GoTemplate ?

### PR DESCRIPTION
**Pull request by s3than**
Fixes #71 

http://masterminds.github.io/sprig/math.html
Functions within the sprig library utilise integer values causing disruption budget to not provide usable values.

### CODE REVIEW

**Comment by mitchellh** [comment]
This is a clever way to do this and good job catching this. Awesome job on the tests too 😮 

I'll defer to @adilyse to hit the button and review 😄

File: test/unit/server-disruptionbudget.bats [hunk 1]
```diff
@@ -55,27 +55,73 @@ load _helpers
...
       -x templates/server-disruptionbudget.yaml  \
       --set 'server.replicas=3' \
       . | tee /dev/stderr |
       yq '.spec.maxUnavailable' | tee /dev/stderr)
-  [ "${actual}" = "0" ]
+  [ "${actual}" = "1" ]
 }
 
-@test "server/DisruptionBudget: correct maxUnavailable with replicas=1" {
+@test "server/DisruptionBudget: correct maxUnavailable with n=4" {
```
Comment (adilyse) [inline test/unit/server-disruptionbudget.bats]:
```suggestion
@test "server/DisruptionBudget: correct maxUnavailable with replicas=4" {
```

File: test/unit/server-disruptionbudget.bats [hunk 2]
```diff
@@ -55,27 +55,73 @@ load _helpers
...
-      --set 'server.replicas=1' \
+      --set 'server.replicas=4' \
       . | tee /dev/stderr |
       yq '.spec.maxUnavailable' | tee /dev/stderr)
-  [ "${actual}" = "0" ]
+  [ "${actual}" = "1" ]
+}
+
+
+@test "server/DisruptionBudget: correct maxUnavailable with n=5" {
```
Comment (adilyse) [inline test/unit/server-disruptionbudget.bats]:
```suggestion
@test "server/DisruptionBudget: correct maxUnavailable with replicas=5" {
```

File: test/unit/server-disruptionbudget.bats [hunk 3]
```diff
@@ -55,27 +55,73 @@ load _helpers
...
+      . | tee /dev/stderr |
+      yq '.spec.maxUnavailable' | tee /dev/stderr)
+  [ "${actual}" = "1" ]
 }
 
-# Note: It is not possible to test anything but the default behavior of the
-# maxUnavailable definition in the _helpers.tpl with the current test setup
-# because the `--set` flag overrides values AFTER they're been run through
-# the helper functions.
+@test "server/DisruptionBudget: correct maxUnavailable with n=6" {
```
Comment (adilyse) [inline test/unit/server-disruptionbudget.bats]:
```suggestion
@test "server/DisruptionBudget: correct maxUnavailable with replicas=6" {
```

File: test/unit/server-disruptionbudget.bats [hunk 4]
```diff
@@ -55,27 +55,73 @@ load _helpers
...
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/server-disruptionbudget.yaml  \
+      --set 'server.replicas=6' \
+      . | tee /dev/stderr |
+      yq '.spec.maxUnavailable' | tee /dev/stderr)
+  [ "${actual}" = "2" ]
+}
+
+@test "server/DisruptionBudget: correct maxUnavailable with n=7" {
```
Comment (adilyse) [inline test/unit/server-disruptionbudget.bats]:
```suggestion
@test "server/DisruptionBudget: correct maxUnavailable with replicas=7" {
```

File: test/unit/server-disruptionbudget.bats [hunk 5]
```diff
@@ -55,27 +55,73 @@ load _helpers
...
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/server-disruptionbudget.yaml  \
+      --set 'server.replicas=7' \
+      . | tee /dev/stderr |
+      yq '.spec.maxUnavailable' | tee /dev/stderr)
+  [ "${actual}" = "2" ]
+}
+
+@test "server/DisruptionBudget: correct maxUnavailable with n=8" {
```
Comment (adilyse) [inline test/unit/server-disruptionbudget.bats]:
```suggestion
@test "server/DisruptionBudget: correct maxUnavailable with replicas=8" {
```

**Comment by s3than** [comment]
@adilyse All updated

**Pull request merged**